common/src/main/java/com/wynntils/core/chat/tabs/ChatTab.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.chat.tabs;
+
+import com.wynntils.core.chat.RecipientType;
+import com.wynntils.mc.event.LocalMessageEvent;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ChatTab {
+    private String name;
+    private boolean lowPriority;
+
+    // Filters
+    private Set<RecipientType> filteredTypes;
+    private String customRegexString;
+
+    private transient Pattern customRegex;
+
+    public ChatTab(String name, boolean lowPriority, Set<RecipientType> filteredTypes, String customRegexString) {
+        this.name = name;
+        this.lowPriority = lowPriority;
+        this.filteredTypes = filteredTypes;
+        this.customRegexString = customRegexString;
```

Comment:
I think it's better to make the field a Pattern and compile it in the constructor, or assign null if no regexp string is given. The `getCustomRegex` looks too much like the failed double-locking pattern.

Reply:
No, you can't do that. Gson won't use the constructor properly. This is more simple than writing a whole deserializer though

Reply:
Ah, are you using gson for the chat tabs? I must have missed that. Why's that? To save history?

Reply:
Configs are saved with gson? Chat Tabs are saved to config in ChatTabFeature.